src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -1144,17 +1145,27 @@ private async Task CreateExecutableAsync(AppResource er, ILogger resourceLogger,
 
             foreach (var arg in args)
             {
-                var value = arg switch
+                try
                 {
-                    string s => s,
-                    IValueProvider valueProvider => await GetValue(key: null, valueProvider, resourceLogger, isContainer: false, cancellationToken).ConfigureAwait(false),
-                    null => null,
-                    _ => throw new InvalidOperationException($"Unexpected value for {arg}")
-                };
+                    var value = arg switch
+                    {
+                        string s => s,
+                        IValueProvider valueProvider => await GetValue(key: null, valueProvider, resourceLogger, isContainer: false, cancellationToken).ConfigureAwait(false),
+                        null => null,
+                        _ => throw new InvalidOperationException($"Unexpected value for {arg}")
+                    };
 
-                if (value is not null)
+                    if (value is not null)
+                    {
+                        spec.Args.Add(value);
+                    }
+                }
+                catch (Exception ex)
                 {
-                    spec.Args.Add(value);
+                    resourceLogger.LogCritical("Failed to apply arguments. A dependency may have failed to start.");
```

Comment:
Why not throw a more specific exception and let this bubble out? That way you can handle it in a single place.

Reply:
Its less interesting for args, but for environment variables I'm trying to log each environment variable that had trouble being injected. I figured a little bit of duplication here led to a better experience. But I could probably capture that in the exception.

Reply:
So you want to keep trying as each one fails instead of failing on the first one?

Reply:
What if we log from here but then throw at the end if any failed to be applied?

Reply:
Fixed.